cli/hclfmt.go

```diff
@@ -0,0 +1,115 @@
+package cli
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/hashicorp/hcl2/hcl"
+	"github.com/hashicorp/hcl2/hclparse"
+	"github.com/hashicorp/hcl2/hclwrite"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// runHCLFmt recursively looks for terragrunt.hcl files in the directory tree starting at workingDir, and formats them
+// based on the language style guides provided by Hashicorp. This is done using the official hcl2 library.
+func runHCLFmt(terragruntOptions *options.TerragruntOptions, workingDir string) error {
+	terragruntOptions.Logger.Printf("Formatting terragrunt.hcl files in the current directory tree.")
+
+	tgHclFiles, err := findTerragruntHclFiles(terragruntOptions, workingDir)
+	if err != nil {
+		return err
+	}
+
+	terragruntOptions.Logger.Printf("Found %d terragrunt.hcl files", len(tgHclFiles))
+
+	for _, tgHclFile := range tgHclFiles {
+		err := formatTgHCL(terragruntOptions, tgHclFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// findTerragruntHclFiles will recursively look for terragrunt.hcl files in the directory tree starting at workingDir.
+func findTerragruntHclFiles(terragruntOptions *options.TerragruntOptions, workingDir string) ([]string, error) {
```

Comment:
Alternative option is to use the [zglob library](https://github.com/mattn/go-zglob) to give back a list in a one-liner.